docs/openstack/manila_adoption.md

```diff
@@ -0,0 +1,326 @@
+
+## Prerequisites
+
+* Ensure that manila systemd services (api, cron, scheduler) are 
+  [stopped](stop_openstack_services.md#stopping-control-plane-services).
+* Ensure that manila pacemaker services ("openstack-manila-share") are 
+  [stopped](stop_openstack_services.md#stopping-control-plane-services).
+* Ensure that the [database migration](mariadb_copy.md) has completed.
+* Ensure that OpenShift nodes where `manila-share` service will be deployed 
+  can reach the management network that the storage system is in.
+* Ensure that services such as [keystone](keystone_adoption.md) 
+  and [memcached](backend_services_deployment.md) are available prior to 
+  adopting manila services.
+* If tenant-driven networking was enabled (`driver_handles_share_servers=True`),
+  ensure that [neutron](neutron_adoption.md) has been deployed prior to 
+  adopting manila services.
+
+## Procedure - Manila adoption
+
+### Copying configuration from the RHOSP 17.1 deployment
+
+Define the `CONTROLLER1_SSH` environment variable, if it [hasn't been 
+defined](stop_openstack_services.md#variables) already. Then copy the 
+configuration file from RHOSP 17.1 for reference.
+
+```
+$CONTROLLER1_SSH cat /var/lib/config-data/puppet-generated/manila/etc/manila/manila.conf | awk '!/^ *#/ && NF' > ~/manila.conf
+```
+
+Review this configuration, alongside any configuration changes that were noted
+since RHOSP 17.1. Not all of it makes sense to bring into the new cloud 
+environment:
+
+<!--- TODO link config diff tables for RHOSP 17.1 (Wallaby) to RHOSP 18 (Antelope) --->
+
+- The manila operator is capable of setting up database related configuration
+  (`[database]`), service authentication (`auth_strategy`,
+  `[keystone_authtoken]`), message bus configuration 
+  (`transport_url`, `control_exchange`), the default paste config
+  (`api_paste_config`) and inter-service communication configuration (`
+  [neutron]`, `[nova]`, `[cinder]`, `[glance]` `[oslo_messaging_*]`). So 
+  all of these can be ignored.
+- Ignore the `osapi_share_listen` configuration. In RHOSP 18, we rely on 
+  OpenShift's routes and ingress.
+- Pay attention to policy overrides. In RHOSP 18, manila ships with a secure 
+  default RBAC, and overrides may not be necessary. Please review RBAC 
+  defaults by using the [Oslo policy generator](https://docs.openstack.org/oslo.policy/latest/cli/oslopolicy-policy-generator.html)
+  tool. If a custom policy is necessary, you must provide it as a 
+  `ConfigMap`. The following sample spec illustrates how a 
+  `ConfigMap` called `manila-policy` can be set up with the contents of a 
+  file called `policy.yaml`.
+
+```yaml
+  spec:
+    manila:
+      enabled: true
+      template:
+        manilaAPI:
+          customServiceConfig: |
+             [oslo_policy]
+             policy_file=/etc/manila/policy.yaml
+        extraMounts:
+        - extraVol:
+          - extraVolType: Undefined
+            mounts:
+            - mountPath: /etc/manila/
```

Comment:
awesome, and this is consistent with what we're doing in glance. +1 on the potential policy.yaml